source/lib.civet

```diff
@@ -2538,6 +2561,50 @@ function processAssignments(statements): void {
     })
 }
 
+function processBlocks(statements): void
+  insertSemicolon(statements)
+  gatherRecursive statements, .type is "BlockStatement"
+  .forEach ({ expressions }) ->
+    processBlocks expressions
+
+/**
+* Avoid automatic continuation onto lines that start with
+* certain characters by adding an explicit semicolon. See
+* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#automatic_semicolon_insertion
+*/
+function insertSemicolon(statements: StatementTuple[]): void
+  l := statements.length
+  statements.forEach (s, i) ->
+    if i < l - 1
+      if needsPrecedingSemicolon(statements[i + 1])
+        delim := s[2]
+        if !delim
+          s[2] = ";"
+        else if typeof delim is "string" and !delim.match(/;/)
+          s[2] = `;${delim}`
```

Comment:
I assume `delim` can be an object, but only when it's already a semicolon so we don't need to do anything? Maybe add a comment, like `// If delim is an object, assume it's already a semicolon`.